qualang_tools/results/results.py

```diff
@@ -68,11 +58,37 @@ def fetch_all(self):
             for data in self.data_list:
                 if hasattr(self.res_handles, data):
                     self.results.append(
-                        self.format_(self.res_handles.get(data).fetch_all())
+                        self._format(self.res_handles.get(data).fetch_all())
                     )
 
         elif self.mode == "live":
             self.results = []
             for i in range(len(self.data_handles)):
-                self.results.append(self.format_(self.data_handles[i].fetch_all()))
+                self.results.append(self._format(self.data_handles[i].fetch_all()))
         return self.results
+
+
+def progress_counter(iteration, total, progress_bar=True, percent=True, start=None):
+    """Displays progress bar and prints remaining computation time.
+
+    :param iteration: current iteration. Must be a python integer.
+    :param total: total number of iteration. Must be a python integer.
+    :param progress_bar: Flag enabling the progress bar display. Must be True or False. Default is True.
+    :param percent: Flag enabling the progress percentage display. Must be True or False. Default is True.
+    :param start: Starting time of the program enabling the elapsed time display. Must be the result of time.time(). Default is None.
+    :return: None.
+    """
+    current_percent = (iteration + 1) / total * 100
+
+    progress = "Progress: "
+    if progress_bar:
+        bar = "#" * int(current_percent / 2)
+        progress += "[" + bar + " " * (50 - len(bar)) + "] "
+    if percent:
+        progress += f"{current_percent:.1f}%"

```

Comment:
If we're rounding it to integers, why display a decimal point?

Reply:
I don't think I am rounding it...